File: Others/PG_크레인인형뽑기게임.py
# 프로그래머스 크레인 인형뽑기 게임
## 스택 개념을 알면 쉽게 푸는 문제

def solution(board, moves):
    answer = 0
    box = []
    board_row = [[] for _ in range(len(board[0]))]
    for i in range(len(board[0])):
        for j in range(len(board) - 1, -1, -1):
            if board[j][i]:
                board_row[i].append(board[j][i])

    for m in moves:
        if len(board_row[m - 1]) == 0:
            continue
        num = board_row[m - 1].pop()
        last_num = box[len(box) - 1] if box else 0
        if num == last_num and last_num:
            box.pop()
            answer += 2
        else:
            box.append(num)
    return answer

File: Others/PG_크레인인형뽑기게임.js
// 프로그래머스 크레인 인형뽑기
function solution(board, moves) {
  let answer = 0;
  let boardRow = []
  let basket = []
  for (let a = 0; a < board[0].length; a++) {
      let temp = []
      for (let b = 0; b < board.length; b++) {
          board[b][a] ? temp.push(board[b][a]) : true
      }
      boardRow.push(temp)
  }
  
  for (let i of moves) {
      if (boardRow[i-1].length) {
          let num = boardRow[i-1].shift()
      
          if (basket.length) {
              if (num === basket[basket.length - 1]) {
                  basket.pop()
                  answer+=2
              } else {
                  basket.push(num)
              }
          } else {
              basket.push(num)
          }
      }
  }
  return answer;
}